student.rb

```diff
@@ -1,12 +1,17 @@
 require_relative 'person'
 
 class Student < Person
-  def initialize(id, age, classroom, name = 'unknown', parent_permission: true)
-    super(id, age, parent_permission, name)
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
     @classroom = classroom
   end
 
   def play_hooky
     '¯\\(ツ)/¯'
   end
+
+  def assign_classroom(classroom)
+    self.classroom = classroom
+    classroom.students.push(self) unless classroom.students.iclude?(self)
+  end
```

Comment:
- [ ] I would suggest that you use a custom `setter` function in this case:

If you look closely to the `add_student` method in the `Classroom` class:

```rb
  def add_student(student)
    student.classroom = self
    @students << student
  end
```

You're using `student.classroom = self`. However, you haven't defined any setter function inside your student class, so this will fail.

```suggestion
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.iclude?(self)
  end
```

Please consider applying this change! 🤓 🔧